espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/InvocationOperation.java

```diff
@@ -0,0 +1,376 @@
+package io.appium.espressoserver.lib.helpers;
+
+import android.util.Log;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import io.appium.espressoserver.lib.model.BackdoorResultVoid;
+
+//https://github.com/calabash/calabash-android-server/blob/develop/server/app/src/androidTest/java/sh/calaba/instrumentationbackend/query/InvocationOperation.java
+
+public class InvocationOperation {
+    private final String methodName;
+    private final List<?> arguments;
+    private final Class<?>[] argumentTypes;
+
+    public InvocationOperation(String methodName, List<?> arguments) {
+        this.methodName = methodName;
+        this.arguments = arguments;
+        this.argumentTypes = parseArgumentTypes(arguments);
+    }
+
+
+    public Object apply(final Object o) throws Exception {
+        Log.d("Backdoor Method", getName());
+        Log.d("Invoking on ", o.toString());
+        final AtomicReference<Object> ref = new AtomicReference<Object>();
+        final AtomicReference<Exception> refEx = new AtomicReference<Exception>();
+
+        InstrumentationRegistry.getInstrumentation().runOnMainSync(new Runnable() {
+            @Override
+            public void run() {
+                MethodWithArguments method = findCompatibleMethod(o);
+
+                if (method == null) {
+                    StringBuilder stringBuilder = new StringBuilder("No such method found: ");
+                    stringBuilder.append(InvocationOperation.this.methodName);
+                    stringBuilder.append("(");
+                    int length = InvocationOperation.this.arguments.size();
+
+                    for (int i = 0; i < length; i++) {
+                        Object argument = InvocationOperation.this.arguments.get(i);
+
+                        if (i != 0) {
+                            stringBuilder.append(", ");
+                        }
+
+                        String simpleName;
+
+                        if (argument == null) {
+                            simpleName = "null";
+                        } else {
+                            simpleName = argument.getClass().getSimpleName();
+                        }
+
+                        stringBuilder.append("[").append(simpleName).append("]");
+                    }
+
+                    stringBuilder.append(")");
+
+                    ref.set(BackdoorResultVoid.instance
+                            .asMap(InvocationOperation.this.methodName, o, stringBuilder.toString()));
+                    return;
+                }
+
+                Object result;
+
+                try {
+                    if (method.getMethod().getReturnType().equals(Void.TYPE)) {
+                        method.invoke(o);
+                        result = "<VOID>";
```

Comment:
what is the purpose of this magic string?

Reply:
If a method with return type "Void" was invoked, then this string will inform the client that method was invoked and results are valid but "<Void>"

Reply:
Please move it to a class-level constant then